File: MyBatis/demos/CustomMyBatis_annotation/src/main/java/com/daliu/mybatis/sqlsession/defaults/DefaultSqlSessionFactory.java
package com.daliu.mybatis.sqlsession.defaults;

import com.daliu.mybatis.cfg.Configuration;
import com.daliu.mybatis.sqlsession.SqlSession;
import com.daliu.mybatis.sqlsession.SqlSessionFactory;

public class DefaultSqlSessionFactory implements SqlSessionFactory {
    private Configuration cfg;
    public DefaultSqlSessionFactory(Configuration cfg) {
        this.cfg = cfg;
    }

    /**
     * 创建一个新的操作数据库对象()
     * @return
     */
    public SqlSession openSession() {
        return new DefaultSqlSession(cfg);
    }
}

File: MyBatis/MyBatis/BMSqliteDemo/XX/DefaultSqlSessionFactory.m
//
//  DefaultSqlSessionFactory.m
//  ProxyDemo
//
//  Created by liuweizhen on 2019/11/3.
//  Copyright © 2019 liuxing8807@126.com. All rights reserved.
//

#import "DefaultSqlSessionFactory.h"
#import "DefaultSqlSession.h"

@interface DefaultSqlSessionFactory ()

@property (nonatomic, strong) Configuration *config;

@end

@implementation DefaultSqlSessionFactory

- (instancetype)initWithConfig:(Configuration *)config {
    if (self = [super init]) {
        self.config = config;
    }
    return self;
}

- (id<SqlSession>)openSession {
    return [[DefaultSqlSession alloc] initWithConfig:self.config];
}

@end
